### ISSUE TITLE
Search by the ID field

### ISSUE THREAD

**Issue by sockmonk** [thoughtbot/administrate#288]
I'd like to a basic search by the actual ID primary key of some of my tables, but there doesn't seem to be an easy way to do that. The best I can do is click on a random record to get its 'show' view, and overwrite the URL.

Maybe the search could be by ID if it's all numeric, or there could be separate "search by id" field? The latter's probably better, especially if people use GUID keys.

**Comment by graysonwright** [comment]
@sockmonk it looks like this is related to #157, #277, and #183.

It probably makes sense to treat the id the same as any other number field, and not have special behavior for it. I'm planning to support searching number fields by `=`, `>`, and `<`, as in:

- `price < 20`,
- `id = 4`,
- `lifetime_value > 200`

At the moment we're focusing on stability and documentation. This will be one of the first features up after we get past that point.

Until then, you might be able to get along in your own app by overriding the `index action` in `app/controllers/admin/application_controller.rb`. Here's [the default implementation][index]:

```ruby
    def index
      search_term = params[:search].to_s.strip
      resources = Administrate::Search.new(resource_resolver, search_term).run
      resources = order.apply(resources)
      resources = resources.page(params[:page]).per(records_per_page)
      page = Administrate::Page::Collection.new(dashboard, order: order)

      render locals: {
        resources: resources,
        search_term: search_term,
        page: page,
      }
    end
```

You can overwrite the line with `Administrate::Search` to add custom searching behavior.

Let me know if that solves your problem! If it does, and if you're happy with the searching roadmap that we're working on, feel free to close the issue.

Cheers!

[index]: https://github.com/thoughtbot/administrate/blob/master/app/controllers/administrate/application_controller.rb#L3-L15

**Comment by sockmonk** [comment]
Thanks for your response, and thanks for a great tool. I tried replacing or adding to the line with Administrate::Search, but couldn't get it to work right. I solved it for now by adding a second search parameter in the view and controller, and in one case a third param, and searching by the new parameter with a straight ActiveRecord call. The CSS is a little off, but it mostly works, and is ok for an internal tool.

Your idea for a number search would work, but I think in my case it would be toooo much typing whenever you wanted to search to the Orders table by "customer_id = 44444" to get a customer's orders.  I'd rather have the "exact match on this field" option, instead of having to "try to guess what field I'm searching for" that is the default.
